### ISSUE TITLE
Enemies try to pounce or charge at their target even if they can't draw a nav path to them

### ISSUE THREAD

**Issue by Talsidor** [CardboardKeep/Warden#128]
This is a problem because: Enemies can get stuck on the other side of a gap or ledge, pouncing or charging on the spot, thinking they're moving towards you but getting blocked by a gap in the nav mesh. They will repeat this indefinitely as long as their pounce/charge trigger conditions remain true, so they're easy to get stuck in a loop.

Repro: One of the best places to see this is the Town Chasm Crossing, where Polto jump between pillars of nav mesh. Standing a few meters away from the edge of one of the platforms often causes the Polto to trigger a pounce or charge at you before they reach the off mesh link jumping point.

History: I tried to implement a fix for this by not trying to initiate a pounce or charge if the nav mesh agent has an upcoming off mesh link (nav.nextOffMeshLinkData.valid), but I'm not confident it made a difference.

**Comment by Agilethief** [comment]
Regarding the history of a check with the nav mesh link, It must have been removed at some point as I can't see it anywhere.

**Comment by Agilethief** [comment]
I added a check for remaining nav distance to the if block asking if they are in range. 
It seems to have resolved the issues. 

However I have found that the polto getting stuck in place on the blocks is a bug with the traverse code. They get trapped in their "AlightTraverse" enumerator and their fwd does not accurately get set to their traverseAlign. 

@Talsidor I move to close this issue and open another issue regarding the traverse stuck bug.
